min and max type definition for MeterHTMLAttributes and ProgressHTMLAttributes are missing number

The type definitions for MeterHTMLAttributes and ProgressHTMLAttributes were missed when fixing the min and max attributes in https://github.com/preactjs/preact/pull/4568.

```
src/components/Progress.tsx:59:48 - error TS2322: Type 'number' is not assignable to type 'Signalish<string | undefined>'.

59         <progress className={classes.progress} max={maxValue}></progress>
                                                  ~~~

  ../node_modules/.pnpm/preact@10.25.1/node_modules/preact/src/jsx.d.ts:1968:3
    1968   max?: Signalish<string | undefined>;
           ~~~
    The expected type comes from property 'max' which is declared here on type 'ProgressHTMLAttributes<HTMLProgressElement>'
```

--- a/src/jsx.d.ts
+++ b/src/jsx.d.ts
@@ -1909,8 +1909,8 @@ export namespace JSXInternal {
 		form?: Signalish<string | undefined>;
 		high?: Signalish<number | undefined>;
 		low?: Signalish<number | undefined>;
-		max?: Signalish<string | undefined>;
-		min?: Signalish<string | undefined>;
+		max?: Signalish<number | string | undefined>;
+		min?: Signalish<number | string | undefined>;
 		optimum?: Signalish<number | undefined>;
 		value?: Signalish<string | number | undefined>;
 	}
@@ -1965,7 +1965,7 @@ export namespace JSXInternal {
 
 	interface ProgressHTMLAttributes<T extends EventTarget>
 		extends HTMLAttributes<T> {
-		max?: Signalish<string | undefined>;
+		max?: Signalish<number | string | undefined>;
 		value?: Signalish<string | number | undefined>;
 	}
 

